app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt

```diff
@@ -863,6 +881,108 @@ class VaultRepositoryImpl(
+            ?: fido2CredentialRequest.callingAppInfo.getAppOrigin()
+
+        val ciphersWithFido2Credentials = ciphersStateFlow.value.data
+            ?.filter { it.isActiveWithFido2Credentials }
+            .orEmpty()
+
+        val registerResult = vaultSdkSource
+            .registerFido2Credential(
+                userId = userId,
+                origin = origin,
+                requestJson = """{"publicKey": ${fido2CredentialRequest.requestJson}}""",
+                clientData = clientData,
+                selectedCipherView = selectedCipherView,
+                cipherViews = ciphersWithFido2Credentials,
+                isVerificationSupported = isVerificationSupported,
+                checkUser = checkUser,
+                checkUserAndPickCredentialForCreation = { options, _ ->
+                    if (options.requireVerification != Verification.DISCOURAGED) {
+                        // TODO [PM-8137]: Trigger user verification as it is preferred|required.
+                        waitUntilUnlocked(userId)
+                    }
+                    CipherViewWrapper(cipher = selectedCipherView)
+                },
+                findCredentials = { fido2CredentialIds, relayingPartyId ->
+                    // We force a sync so that the SDK has the latest version of any ciphers that
+                    // contain a matching credential.
+                    sync()
+                    vaultSdkSource
+                        .decryptFido2CredentialAutofillViews(
+                            userId = userId,
+                            cipherViews = ciphersWithFido2Credentials.toTypedArray(),
+                        )
+                        .map { decryptedFido2CredentialViews ->
+                            decryptedFido2CredentialViews.filter { fido2CredentialView ->
+                                fido2CredentialView.rpId == relayingPartyId &&
+                                    fido2CredentialIds.contains(fido2CredentialView.credentialId)
+                            }
+                        }
+                        .map { matchingFido2Credentials ->
+                            ciphersWithFido2Credentials.filter { cipherView ->
+                                matchingFido2Credentials.any { it.cipherId == cipherView.id }
+                            }
+                        }
+                        .fold(
+                            onSuccess = { matchingCipherViews ->
+                                FindFido2CredentialsResult.Success(
+                                    cipherViews = matchingCipherViews,
+                                )
+                            },
+                            onFailure = { FindFido2CredentialsResult.Error },
+                        )
+                },
+                saveCipher = { cipher ->
+                    vaultSdkSource
+                        .decryptCipher(userId, cipher)
+                        .map { createCipher(it) }
```

Comment:
🤔 Are you going to implement `updateCipher(it)` on the flow where the user chooses an existing cipher from the list to register the Fido2 credential in another PR? Basically, here it should check whether the cipher has an `id` to **create** or **update** it (maybe there's already an existing method that performs this checks and guides the flow towards the proper action)

Reply:
I can actually add that here with a minor change. Good callout!